### PR TITLE
nixos/movim: init, movim: additions + patches

### PR DESCRIPTION
**Pull request by toastal**
## Description of changes

Added Movim as a service. This includes a few awaiting-review patches to upstream on Movim’s Microsoft GitHub proprietary code forge--some are for making NixOS work better & I needed to move to the latest commit to avoid the patches conflicts for:

* ~~https://github.com/movim/movim/pull/1280~~
* https://github.com/movim/movim/pull/1281
* https://github.com/movim/movim/pull/1282
* https://github.com/movim/movim/pull/1284.
* https://github.com/movim/movim/pull/1285
* https://github.com/movim/movim/pull/1286
* https://github.com/movim/movim/pull/1287
* https://github.com/movim/movim/pull/1288
* https://github.com/movim/movim/pull/1290

Additionally:

* A few paths needed patching
* ImageMagick patch fixes were closed since 6.x is current support, but Nix will be using 7.x

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

[^1]

[^1]: Please consider [giving up MS GitHub](https://sfconservancy.org/GiveUpGitHub/) or offering a non-proprietary, non-US-corporate-controlled mirror for this free software project.
I wish to delete this Microsoft account in the future, but I need more projects like this to support alternative methods to send patches & contribute.

### CODE REVIEW

**Comment by Lassulus** [comment]
yoyo, is this ready for merging? I guess we can also fix stuff later?

File: nixos/modules/services/web-apps/movim.nix [hunk 1]
```diff
@@ -0,0 +1,711 @@
...
+      };
+
+      tmpfiles.rules = with cfg; [
+        "d ${cfg.dataDir}/             0710 ${user} ${group} - -"
+        "d ${cfg.dataDir}/public       0750 ${user} ${group} - -"
+        "d ${cfg.dataDir}/public/cache 0750 ${user} ${group} - -"
+        "d ${cfg.runtimeDir}/          0700 ${user} ${group} - -"
+        "d ${cfg.runtimeDir}/cache     0700 ${user} ${group} - -"
+        "d ${cfg.logDir}               0700 ${user} ${group} - -"
+      ];
```
Comment (Lassulus) [inline nixos/modules/services/web-apps/movim.nix:708]: there is tmpfiles.settings now, which allows for a more nix like syntax

File: nixos/modules/services/web-apps/movim.nix [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+        };
+      };
+
+      services.${phpExecutionUnit} = {
+        after = [ "movim-data-setup.service" ];
+        requires = [ "movim-data-setup.service" ]
+          ++ lib.optional cfg.database.createLocally dbService;
+      };
+
+      tmpfiles.settings = with cfg; {
```
Comment (Lassulus) [inline nixos/modules/services/web-apps/movim.nix:701]:
```suggestion
      tmpfiles.settings.movil = with cfg; {
```
you need to specify the filename/namespace also

**Comment by Neustradamus** [comment]
@toastal: Good job!

**Comment by Neustradamus** [comment]
@toastal: I do not use your code but it is a good news for XMPP users in the World :)

**Pull request merged**